rust/xaynet-client/src/api/http.rs

```diff
@@ -21,13 +23,68 @@ pub struct HttpApiClient {
 }
 
 impl HttpApiClient {
-    pub fn new<S>(address: S) -> Self
+    /// Creates a new HTTP(S) client.
+    ///
+    /// # Warning
+    /// If no trusted root `certificate`s are provided, then every server will be trusted.
+    pub fn new<S>(
+        address: S,
+        certificates: Option<Vec<Certificate>>,
+    ) -> Result<Self, HttpApiClientError>
     where
         S: Into<String>,
     {
-        Self {
-            client: Client::new(),
+        let client = if let Some(certificates) = certificates {
+            let mut builder = ClientBuilder::new().use_rustls_tls();
+            for certificate in certificates {
+                builder = builder.add_root_certificate(certificate);
+            }
+            builder.build().map_err(HttpApiClientError::Http)?
+        } else {
+            ClientBuilder::new()
+                .use_rustls_tls()
+                .danger_accept_invalid_certs(true)
```

Comment:
No sure if I understand that `else` branch correctly. 
If we don't use certificated, we allow invalid certificates? 
Can't we just say: if no certificate is provided, only http is supported?

Reply:
this issue comes from the fact that `reqwest` now has the `tls` feature enabled, which causes the `Client` to have a vector of certificates. i *assume*  that every connection will fail if this vector is empty, therefore i set it to accept everything (as it would if there was no tls enabled).

i agree that this is somewhat suboptimal and potentially confusing. a better solution would be to introduce a feature flag for tls ourselves and to gate the respective parts behind config attributes. this would also allow us to remove the `Option<>`s around the certificate paths to clean up some code. should we do so right away or in a future ticket?

Reply:
Good point. A feature flag on the client side would make sense I think👍 
> should we do so right away or in a future ticket? 

I'm fine if we do it later.

Reply:
i couldn't let go, so it is part of this pr now as well :D looks nicer imo